incubator/openldap/templates/openldap-deployment.yaml

```diff
@@ -0,0 +1,78 @@
+#
+# A Deployment spec for openldap slapd
+#
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name:  {{ template "openldap.fullname" . }}
+  labels:
+    app: {{ template "openldap.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "openldap.name" . }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "openldap.name" . }}
+        release: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: {{ .Values.containerName | quote }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+        ports:
+        - name: ldap
+          containerPort: {{ .Values.service.ldapInternalPort }}
```

Comment:
It probably doesn't make sense to make container ports configurable. Change them will break things. I'd suggest you hard-code them, give them names, and reference them via their names from service and probes (which you should add).